### PR TITLE
Add thermal conduction to the driver

### PR DESCRIPTION
**Pull request by jamiebramwell**
This adds thermal conduction options for the driver code. It also allows the `ThermalSolid` module to be constructed via an input file.

### CODE REVIEW

File: src/drivers/serac.cpp [hunk 1]
```diff
@@ -97,12 +100,32 @@ int main(int argc, char* argv[])
...
-  // Define the solid solver object
-  auto                  solid_solver_options = inlet["nonlinear_solid"].get<serac::NonlinearSolid::InputOptions>();
-  serac::NonlinearSolid solid_solver(mesh, solid_solver_options);
+  // Create the physics object
+  std::unique_ptr<serac::BasePhysics> main_physics;
+
+  std::optional<serac::NonlinearSolid::InputOptions>    solid_solver_options;
+  std::optional<serac::ThermalConduction::InputOptions> thermal_solver_options;
+
+  if (inlet.contains("nonlinear_solid")) {
```
Comment (white238) [inline src/drivers/serac.cpp:109]: What if it contains both?

File: src/drivers/serac.cpp [hunk 2]
```diff
@@ -97,12 +100,32 @@ int main(int argc, char* argv[])
...
+    solid_solver_options = inlet["nonlinear_solid"].get<serac::NonlinearSolid::InputOptions>();
+  }
+
+  if (inlet.contains("thermal_conduction")) {
+    thermal_solver_options = inlet["thermal_conduction"].get<serac::ThermalConduction::InputOptions>();
+  }
+
+  if (solid_solver_options && thermal_solver_options) {
```
Comment (white238) [inline src/drivers/serac.cpp:128]: 👍

File: src/drivers/serac.cpp [hunk 3]
```diff
@@ -113,7 +136,7 @@ int main(int argc, char* argv[])
 
   // FIXME: This and the FromInlet specialization are hacked together,
   // should be inlet["output_type"].get<OutputType>()
-  solid_solver.initializeOutput(inlet.getGlobalTable().get<serac::OutputType>(), "serac");
+  main_physics->initializeOutput(inlet.getGlobalTable().get<serac::OutputType>(), "serac");
 
   // enter the time step loop. This was modeled after example 10p.
```
Comment (white238) [inline src/drivers/serac.cpp]: Is this comment still relevant?

File: src/serac/physics/thermal_solid.hpp [hunk 4]
```diff
@@ -36,6 +36,16 @@ class ThermalSolid : public BasePhysics {
   ThermalSolid(int order, std::shared_ptr<mfem::ParMesh> mesh, const ThermalConduction::SolverOptions& therm_options,
                const NonlinearSolid::SolverOptions& solid_options);
 
+  /**
+   * @brief Construct a new Thermal Solid object from input deck options
```
Comment (white238) [inline src/serac/physics/thermal_solid.hpp]: Trying to move away from "input deck", please use "input file".  I know I keep doing it myself.

**Review by white238 — APPROVED** [review]

File: src/drivers/serac.cpp (hunk 3 above)
Comment (jamiebramwell) [reply src/drivers/serac.cpp]: No idea. Any thoughts @joshessman-llnl ?

Comment (white238) [reply src/drivers/serac.cpp]: I was more referring to the "This was modeled after example 10p" part.

File: src/serac/physics/thermal_solid.hpp (hunk 4 above)
Comment (jamiebramwell) [reply src/serac/physics/thermal_solid.hpp]: BUT I'M OLDDDDD.

Comment (white238) [reply src/serac/physics/thermal_solid.hpp]: =( it's so hard....

File: src/drivers/serac.cpp (hunk 3 above)
Comment (jamiebramwell) [reply src/drivers/serac.cpp]: Heh. Whoops. And the answer is no. It should be fixed now!

**Review by joshessman-llnl — APPROVED** [review]

**Comment by jamiebramwell** [comment]
LGTM

**Pull request merged**